File: backend/schemas/activity.py
from datetime import date
from typing import List, Literal, Optional
from pydantic import BaseModel, field_validator

ActivityType = Literal[
    "wedding", "mehandi", "garba", "haldi", "reception",
    "puja", "sangeet", "engagement", "tilak", "griha_pravesh",
    "naamkaran", "custom",
]
ActivityStatus = Literal["upcoming", "active", "completed"]


class ActivityCreate(BaseModel):
    name: str
    type: ActivityType
    custom_type_name: Optional[str] = None
    date: date
    time: Optional[str] = None
    description: Optional[str] = None
    guest_ids: List[str] = []

    @field_validator("name")
    @classmethod
    def strip_name(cls, v: str) -> str:
        return v.strip()


class ActivityUpdate(BaseModel):
    name: Optional[str] = None
    type: Optional[ActivityType] = None
    custom_type_name: Optional[str] = None
    date: Optional[date] = None
    time: Optional[str] = None
    description: Optional[str] = None
    guest_ids: Optional[List[str]] = None


class ActivityStatusUpdate(BaseModel):
    status: ActivityStatus


class GuestIdsRequest(BaseModel):
    guest_ids: List[str]

File: frontend/types/activity.ts
export type ActivityType =
  | "wedding"
  | "mehandi"
  | "garba"
  | "haldi"
  | "reception"
  | "puja"
  | "sangeet"
  | "engagement"
  | "tilak"
  | "griha_pravesh"
  | "naamkaran"
  | "custom"

export type ActivityStatus = "upcoming" | "active" | "completed"

export interface Activity {
  id: string
  event_id: string
  name: string
  type: ActivityType
  custom_type_name?: string
  date: string
  time?: string
  description?: string
  status: ActivityStatus
  guest_ids: string[]
  guest_count: number
  entry_count: number
  entry_total: number
  created_at: string
  updated_at: string
}

export interface ActivityDetail extends Activity {
  guests: {
    id: string
    name: string
    phone: string
    village: string
    relation_side: string
    rsvp_status: string
  }[]
  entry_summary: {
    total_amount: number
    entry_count: number
    upi_amount: number
    cash_amount: number
  }
}

export const ACTIVITY_TYPE_LABELS: Record<ActivityType, string> = {
  wedding: "Wedding (Vivah)",
  mehandi: "Mehndi Ceremony",
  garba: "Garba Night",
  haldi: "Haldi Ceremony",
  reception: "Reception",
  puja: "Puja",
  sangeet: "Sangeet",
  engagement: "Engagement",
  tilak: "Tilak",
  griha_pravesh: "Griha Pravesh",
  naamkaran: "Naamkaran",
  custom: "Custom",
}

export const ACTIVITY_STATUS_COLORS: Record<ActivityStatus, string> = {
  upcoming: "bg-blue-100 text-blue-700",
  active: "bg-green-100 text-green-700",
  completed: "bg-gray-100 text-gray-600",
}
